### DISCUSSION TITLE
How to get HTML of every pages at once ?

### DISCUSSION THREAD

**Discussion by echobinod** [GrapesJS/grapesjs#3684] [Q&A] [answered]
I can get HTML of each page using pageId this way.
`const pageManager = Editor.Pages;`
`const html = pageManager.get("pageId").getMainFrame().getComponent.getEl();`

I have to select the page to get it's HTML.
`pageManager.select(pageId)`

I don't want to select the page to get its HTML because it switch pages on canvas.

**Comment by artf** [comment, marked as answer]
You can do this
```js
const allPages = editor.Pages.getAll();
const htmlAll = allPages.map(p => p.getMainComponent().toHTML())
```

**Reply by gxanshu** [reply]
and how to get CSS of all pages?

**Reply by lexoyo** [reply]
Maybe you can use [what I did here](https://github.com/silexlabs/Silex/blob/dev/src/ts/client/grapesjs/publish.ts#L325)?

```js
const component = page.getMainComponent()
const html = editor.getHtml({ component })
const css = editor.getCss({ component })
```

Not sure how different it is from the `.toHTML()` way?

**Reply by gxanshu** [reply]
Thanks, it works ✨

**Comment by sridharK64** [comment]
@artf  allPages.map(p => p.getMainComponent().toHTML()) by using this we only getting HTM, is there any way to get Css with it, for example allPages.map(p => p.getMainComponent().toCSS())

const component = page.getMainComponent()
const html = editor.getHtml({ component })
const css = editor.getCss({ component })
in this way we only getting current page html & css

**Reply by mosh-tudor** [reply]
@sridharK64 

```js
const allPages = editor.Pages.getAll();
const htmlAll = allPages.map(p => p.getMainComponent().toHTML());
const cssAll = allPages.map(p => editor.getCss({ p.getMainComponent() }));
```

**Reply by sridharK64** [reply]
Thank you very much for responding
const cssAll = allPages.map(p => editor.getCss({ p.getMainComponent() })); -> by using this getting only Main page or first page css repeatedly.
as @artf shared above code is working for HTML, not working to get css  
const cssAll = allPages.map(p => editor.getCss({ p.getMainComponent() })); there is syntax error, i think we need to use editor.getCss(p.getMainComponent() )
plz refer the SS

![image](https://github.com/GrapesJS/grapesjs/assets/159991191/334f320b-9a2e-404b-93e6-fed43f1eef13)

**Reply by lexoyo** [reply]
`{ p.getMainComponent() }` probably needs to be `{ component: p.getMainComponent() }`

**Reply by sridharK64** [reply]
Thank you very much, @lexoyo. It is working as expected. Have a nice day!

**Reply by sridharK64** [reply]
Hi,
there is a issue with grapesjs-tailwind & grapesjs-plugin-ckeditor, when i was using both together i unable to use grapesjs-plugin-ckeditor options like pointes, font size etc,

here i provided the SS
1. when tailwind installed
![image](https://github.com/GrapesJS/grapesjs/assets/159991191/d96f9b05-57e3-40e7-9677-93cb76fc910c)
2. with out tailwind css
![image](https://github.com/GrapesJS/grapesjs/assets/159991191/03e1d282-930c-4e54-8148-32402665f358)

nothing is changes,

also when i was using grapesjs-plugin-ckeditor, i unable to edit link when we drag & drop

**Reply by artf** [reply]
@sridharK64 please avoid creating a big off-topic discussion. Create a new discussion if it's necessary.
